web/js/precallView.js

```diff
@@ -0,0 +1,236 @@
+    }
+    this.render = function (aData) {
+      return this._templatePromise.then(function (aTemplate) {
+        return aTemplate(aData);
+      });
+    };
+  };
+
+  var init = function () {
+    return new Promise(function (resolve) {
+      if (alreadyInitialized) {
+        return resolve();
+      }
+
+      Utils.addEventsHandlers('', eventHandlers);
+      _template = new exports.EJS({ url: _templateSrc });
+      alreadyInitialized = true;
+      return render(resolve);
+    });
+  };
+
+  var hide = function () {
+    document.getElementById('video-preview').style.visibility = 'hidden';
+    document.getElementById('dock').style.visibility = 'visible';
+  };
+
+  var setVolumeMeterLevel = function (level) {
+    document.getElementById('audio-meter-level').style.width = (level * 89) + 'px';
+  };
+
+  var startPrecallTestMeter = function () {
+    var TEST_DURATION_MAX = 200; // 20 seconds
+    var meterLevel = document.getElementById('precall-test-meter-level');
+    setSwitchStatus(true, false, 'Video', 'roomView:initialVideoSwitch');
+    document.getElementById('test-status').innerText = 'Testing audio / video quality…';
+    meterLevel.style.width = 0;
+    meterLevel.style['animation-play-state'] = 'running';
+    var preCallTestProgress = 0;
+    testMeterInterval = setInterval(function () {
+      preCallTestProgress++;
+      setTestMeterLevel(preCallTestProgress / TEST_DURATION_MAX);
+      if (preCallTestProgress === TEST_DURATION_MAX) {
+        clearInterval(testMeterInterval);
+      }
+    }, 100);
+  };
+
+  var setTestMeterLevel = function (value) {
+    var width = value * document.getElementById('precall-test-meter').offsetWidth;
+    document.getElementById('precall-test-meter-level').style.width = width + 'px';
+  };
+
+  var displayNetworkTestResults = function (results) {
+    var packetLossStr;
+    var audioResultsElem = document.getElementById('precall-audio-results');
+    var videoResultsElem = document.getElementById('precall-video-results');
+
+    clearInterval(testMeterInterval);
+    document.getElementById('test-status').innerText = 'Done.';
+    document.getElementById('precall-test-meter-level').style['animation-play-state'] = 'paused';
+    setTestMeterLevel(1);
+
+    document.getElementById('pre-call-test-results').style.display = 'block';
+    document.getElementById('audio-bitrate').innerText =
+      Math.round(results.audio.bitsPerSecond / 1000);
+    if (results.video && results.text !== 'Your bandwidth can support audio only.') {
+      document.getElementById('video-bitrate').innerText =
+        Math.round(results.video.bitsPerSecond / 1000);
+      packetLossStr = isNaN(results.video.packetLossRatio) ? '' :
+        Math.round(100 * results.video.packetLossRatio) + '% packet loss';
+      document.getElementById('precall-video-packet-loss').innerText = packetLossStr;
+      audioResultsElem.style.width = '50%';
```

Comment:
Again wouldn't this be better off in the .less?

Reply:
Actually, we don't need this CSS adjustment anymore. We always display both the Audio and Video results (with video saying "no video" if there is no video).

Reply:
Awesome 👍